nautobot/ipam/migrations/0031_ipam__prefix__add_parent.py

```diff
@@ -43,6 +46,77 @@ def reverse_it(apps, schema_editor):
         vrf.delete()
 
 
+def migrate_ipaddress_status_to_type(apps, schema_editor):
+    """Migrate IPAddress.status DHCP/SLAAC to types."""
+    Status = apps.get_model("extras", "Status")
+    IPAddress = apps.get_model("ipam", "IPAddress")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    ipaddress_ct = ContentType.objects.get_for_model(IPAddress)
+    statuses = Status.objects.filter(content_types=ipaddress_ct)
+    status_migrated, _ = statuses.get_or_create(
+        name="Migrated",
+        defaults={
+            "color": "ff0000",
+            "description": "DHCP/SLAAC status replaced with `type` of same name by Nautobot 2.0 data migrations.",
+        },
+    )
+
+    # Update all objects of status=DHCP to type=DHCP & status=Migrated.
+    if "test" not in sys.argv:
+        print(">>> Migrating IPAddresses with status DHCP to type DHCP...")
+
+    IPAddress.objects.filter(status__name="DHCP").update(
+        type=nautobot.ipam.choices.IPAddressTypeChoices.TYPE_DHCP,
+        status=status_migrated,
+    )
+
+    # Update all objects of status=SLAAC to type=SLAAC & status=Migrated.
+    if "test" not in sys.argv:
+        print(">>> Migrating IPAddresses with status SLAAC to type SLAAC...")
+
+    IPAddress.objects.filter(status__name="SLAAC").update(
+        type=nautobot.ipam.choices.IPAddressTypeChoices.TYPE_SLAAC,
+        status=status_migrated,
+    )
+
+    # Delete the legacy status objects.
+    print(">>> Deleting Status DHCP")
+    try:
+        Status.objects.filter(name="DHCP").delete()
+    except models.ProtectedError:
+        pass
```

Comment:
How about:

```suggestion
    try:
        status_dhcp = Status.objects.get(name="DHCP")
        status_dhcp.content_types.remove(ipaddress_ct)
        if not status_dhcp.content_types.exists():
            print(">>> Deleting Status DHCP")
            status_dhcp.delete()
    except Status.DoesNotExist:
        pass
```

Same suggestion below.